src/System.Net.WebSockets.Client/tests/CancelTest.cs

```diff
@@ -130,14 +129,9 @@ public class CancelTest : ClientWebSocketTestBase
                 var recvBuffer = new byte[100];
                 var segment = new ArraySegment<byte>(recvBuffer);
 
-                try
-                {
-                    await cws.ReceiveAsync(segment, cts.Token);
-                    Assert.True(false, "Receive should not complete.");
-                }
-                catch (OperationCanceledException) { }
-                catch (ObjectDisposedException) { }
-                catch (WebSocketException) { }
+                Exception e = await Assert.ThrowsAnyAsync<Exception>(() => cws.ReceiveAsync(segment, cts.Token));
+                Assert.True(e is OperationCanceledException || e is ObjectDisposedException || e is WebSocketException,
+                    $"Unexpected exception type {e.GetType()}");
```

Comment:
If this does fail, we end up losing data about the exception, e.g. its stack trace.  Makes me wonder if, while not ideal, the original form that was using Assert.True is better for this case, since there at least the exception is allowed to propagate if it doesn't match the criteria / catch blocks.  At the very least if we do want to keep this as an Assert.True, the message should be changed to something like ```$"Unexpected exception: {e}"``` rather than limiting it to its type.